lib/devicetrust/native/device_windows.go

```diff
@@ -494,12 +498,25 @@ func solveTPMEnrollChallenge(
 	if encryptedCredential == nil {
 		return nil, trace.BadParameter("missing encrypted credential in challenge from server")
 	}
-	activationSolution, err := ak.ActivateCredential(
-		tpm,
-		*encryptedCredential,
-	)
-	if err != nil {
-		return nil, trace.Wrap(err, "activating credential with challenge")
+
+	var activationSolution []byte
+	if windows.GetCurrentProcessToken().IsElevated() {
+		log.Debug("TPM: Detected current process is elevated. Will run credential activation in current process.")
+		// If we are running with elevated privileges, we can just complete the
+		// credential activation here.
+		activationSolution, err = ak.ActivateCredential(
+			tpm,
+			*encryptedCredential,
+		)
+		if err != nil {
+			return nil, trace.Wrap(err, "activating credential with challenge")
+		}
+	} else {
+		log.Info("TPM: Detected that the current process is not running with elevated privileges. Triggering a UAC prompt to run the credential activation in an elevated process.")
```

Comment:
It's probably a good idea to print something to the console too, in case the person misses the UAC popup, or is using an environment where it is not visible.

For reference, touch ID prints this: https://github.com/gravitational/teleport/blob/0bd25172a6fcc918f11cb501c1aa1e0790c0bea4/lib/auth/touchid/api.go#L47-L48

Reply:
I'd completely forgotten that `tsh` doesn't show any logs unless `-d` is presented. I'll add a `fmt.Println`.

Reply:
f3be24c578